src/main/java/dev/openfga/sdk/errors/FgaError.java

```diff
@@ -0,0 +1,47 @@
+package dev.openfga.sdk.errors;
+
+import static dev.openfga.sdk.errors.HttpStatusCode.*;
+
+import java.net.http.HttpHeaders;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+
+public class FgaError extends ApiException {
+    public FgaError(String message, Throwable cause, int code, HttpHeaders responseHeaders, String responseBody) {
+        super(message, cause, code, responseHeaders, responseBody);
+    }
+
+    public FgaError(String message, int code, HttpHeaders responseHeaders, String responseBody) {
+        super(message, code, responseHeaders, responseBody);
+    }
+
+    public static Optional<FgaError> getError(String name, HttpResponse<String> response, Throwable previousError) {
+        int status = response.statusCode();
+
+        // FGA and OAuth2 servers are only expected to return HTTP 2xx responses.
+        if (isSuccessful(status)) {
+            return Optional.empty();
+        }
+
+        final String body = response.body();
+        final var headers = response.headers();
+        final FgaError error;
+
+        if (status == BAD_REQUEST || status == UNPROCESSABLE_ENTITY) {
```

Comment:
What's the baseline Java version for this SDK? If you can use the new `switch()` expression, you might be able to clean this up a bit. https://blogs.oracle.com/javamagazine/post/new-switch-expressions-in-java-12

Reply:
Java 11

(With apologies to anyone still using JDK 8)